### PR TITLE
Query Tracker Engine Liveness odin check logging improvement

### PR DESCRIPTION
**Pull request by Krock21**
Currently, for long queries we have spam in check report:
```
UTC+0000 14:25:39.985 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
UTC+0000 14:25:41.84 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
UTC+0000 14:25:42.163 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
UTC+0000 14:25:43.242 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
UTC+0000 14:25:44.328 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
UTC+0000 14:25:45.408 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
UTC+0000 14:25:46.497 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
UTC+0000 14:25:47.579 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
UTC+0000 14:25:48.658 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
UTC+0000 14:25:49.755 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
UTC+0000 14:25:50.841 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
UTC+0000 14:25:51.922 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
UTC+0000 14:25:52.997 - INFO : 25bca15f	Query 1334ba2c-dfd59531-484e3a5c-849b9df1: running
```

Removing this spam here

### CODE REVIEW

**Comment by Kontakter** [comment]
Maybe it is better to increase logging period?

**Comment by Krock21** [comment]
> Maybe it is better to increase logging period?

An YQL query execution time is somewhere between 1 second and 205 seconds. I would prefer to see in logs exact time when query state was changed

Larger logging period will make it worse(exponential backoff included)

**Comment by Kontakter** [comment]
Ok, in this case I think you could remove `current_time - last_log_time >= LOG_FREQUENCY` and drop `LOG_FREQUENCY` completely.

**Comment by Kontakter** [comment]
Also let wait comments from @Krisha11

File: yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py [hunk 1]
```diff
@@ -41,13 +40,15 @@ def log_query(logger, query_id, state):
...
 
         current_time = time.time()
-        if current_time - last_log_time >= LOG_FREQUENCY:
+        if state != last_state:
             log_query(logger, query_id, state)
             last_log_time = current_time
+            last_state = state
 
         if check_terminal_state(state):
             return state
```
Comment (Krisha11) [inline yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py:54]: Is it true that the exact time is so important - that you need to log more often than once per second?

Comment (Krock21) [reply yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py:54]: It is no so important. Once per second is a good balance

File: yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py [hunk 2]
```diff
@@ -13,7 +13,6 @@ def __init__(self, schema, source_data, result_data, dynamic=False):
 
 
 POLL_FREQUENCY = 0.1
```
Comment (Krock21) [inline yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py:15]:
```suggestion
POLL_FREQUENCY = 1.0
```

File: yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py (hunk 1 above)
Comment (Krock21) [reply yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py:54]: Changed poll frequency from 0.1 to 1.0 seconds

**Review by Krisha11 — APPROVED** [review]

**Comment by Krisha11** [comment]
```
yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py:50: [F841] local variable 'last_log_time' is assigned to but never used
```

File: yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py [hunk 3]
```diff
@@ -41,13 +40,15 @@ def log_query(logger, query_id, state):
 
 def track_query(yt_client, logger, query_id, stage):
     last_log_time = time.time()
+    last_state = ""
```
Comment (Krock21) [inline yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py:43]:
```suggestion
    last_state = ""
```

File: yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py [hunk 4]
```diff
@@ -41,13 +40,15 @@ def log_query(logger, query_id, state):
...
     while True:
         state = get_query_state(yt_client, query_id, stage)
 
         current_time = time.time()
-        if current_time - last_log_time >= LOG_FREQUENCY:
+        if state != last_state:
             log_query(logger, query_id, state)
             last_log_time = current_time
+            last_state = state
```
Comment (Krock21) [inline yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py:51]:
```suggestion
            last_state = state
```

File: yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py [hunk 5]
```diff
@@ -40,14 +39,14 @@ def log_query(logger, query_id, state):
 
 
 def track_query(yt_client, logger, query_id, stage):
-    last_log_time = time.time()
+    last_state = ""
     while True:
         state = get_query_state(yt_client, query_id, stage)
 
         current_time = time.time()
```
Comment (Krock21) [inline yt/odin/checks/lib/query_tracker_engine_liveness/__init__.py:46]:
```suggestion

```

**Review by Krisha11 — APPROVED** [review]

**Pull request closed without merging**